R/data-combined.R

```diff
@@ -0,0 +1,867 @@
+#' @param dataSets Instance (or a `list` of instances) of the `DataSet`
+#'   class.
+#' @param groups A string or a list of strings assigning the data set to a
+#'   group. If an entry within the list is `NULL`, the corresponding data set is
+#'   not assigned to any group (and the corresponding entry in the dataframe
+#'   will be an `NA`). If provided, `groups` must have the same length as
+#'   `dataSets` and/or `simulationResults`. If no grouping is specified for any
+#'   of the dataset, the column `group` in the dataframe output will be all
+#'   `NA`.
+#' @param names A string or a list of string. This argument will be encountered
+#'   across different methods:
+#'
+#' - `$setDataTransformations()`: In the context of this method, a list of names
+#' specifying which observed datasets and/or paths in simulated dataset to
+#' transform with the specified transformations. Default is `NULL`, i.e., the
+#' transformations, if any specified, will be applied to all rows of the
+#' dataframe.
+#'
+#' - `$addSimulationResults()`: In the context of this method, a list of strings
+#' assigning new names to the quantities or paths present in the entered
+#' `SimulationResults` object. Note that the datasets whose names you wish to
+#' not change should be specified as `NULL` in the list.
+#'
+#' - `$addDataSets()`: In the context of this method, a list of strings
+#' assigning new names to the list of instances of the `DataSet` class. Note
+#' that the datasets whose names you wish to not change should be specified as
+#' `NULL` in the list.
+#'
+#'
+#' @import tidyr
+#' @import ospsuite.utils
+#'
+#' @examples
+#'
+#' # load the simulation
+#' simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simFilePath)
+#' simulationResults <- runSimulations(simulations = sim)
+#'
+#' # create a new dataset object
+#' dataSet <- DataSet$new()
+#'
+#' # created object with datasets combined
+#' myCombDat <- DataCombined$new()
+#' myCombDat$addSimulationResults(simulationResults)
+#' myCombDat$addDataSets(dataSet)
+#'
+#' # print the object
+#' myCombDat
+#' @docType class
+#' @export
+
+DataCombined <- R6::R6Class(
+  classname = "DataCombined",
+  inherit = Printable,
+
+  # public fields and methods ------------------------------------
+
+  public = list(
+
+    ## constructor method ---------------
+
+    #' @description
+    #' Initialize a new instance of the class.
+    #'
+    #' @return A new (empty) instance of `DataCombined` class.
+
+    # empty constructor
+    initialize = function() {
+    },
+
+    ## setter methods ---------------
+
+    #' @description
+    #' Adds observed data.
+    #' @return `DataCombined` object containing observed data.
+
+    addDataSets = function(dataSets, names = NULL, groups = NULL) {
+      # validate the object type (`NULL` elements are not accepted)
+      # if list of instances is provided, `purrr::walk` will check each element
+      if (is.list(dataSets)) {
```

Comment:
Are you sure you are looking at the latest version of the files?

Because what you are referring to was present in the files a couple of days ago. Since then I have changed the file and we no longer allow anything in a list that is not a DataSet class instance.

Reply:
oh I see. But validateIsOfType accepts a list as well as parameter. SO I think you don't need the "walk" at all in that case ? Or am I mistaken?

Reply:
> validateIsOfType accepts a list as well as parameter

I don't think so. I've already created an issue about this:
https://github.com/Open-Systems-Pharmacology/OSPSuite.RUtils/issues/21

It will be nice if it could also handle a list.